File: src/libs/cpp/functions/merge.cpp
#include "merge.hpp"

void merge(Context &context) noexcept
{
	// Concatenate two arrays
	auto params = std::get<std::vector<Value>>(context.stack.pop());
	auto array1 = params[0].to_array();
	auto array2 = params[1].to_array();

	array1.reserve(array1.size() + array2.size());
	array1.insert(array1.end(), array2.begin(), array2.end());

	context.stack.push(array1);
}

File: src/runtime/functions/merge.lua
return function(vm)
	local v = vm.pop()
	if type(v[1]) ~= 'table' then v[1] = { v[1] } end
	if type(v[2]) ~= 'table' then v[2] = { v[2] } end

	for i = 1, #v[2] do
		table.insert(v[1], v[2][i])
	end
	vm.push(v[1])
end
